ecosystem/sep-0010.md

```diff
@@ -139,7 +139,81 @@ operation of the challenge is the **Home Domain**, and that the web auth domain
 GET <WEB_AUTH_ENDPOINT>
 ```
 
-Request Parameters:
+##### Request Header:
+
+Optionally, the server may forbid unauthorized calls to `GET <WEB_AUTH_ENDPOINT>`. This could be done to protect an
+endpoint against bad actors or limit access only to allowed applications.
+
+In that case, the client must add an `Authorization` header, and the server should respond to all requests with missing
+header with an error (see below). Server may choose to only accept calls from trusted applications, responding with an
+error to all other requests (see below).
+
+Authorized `WEB_AUTH_ENDPOINT` accepts `Authorization Bearer: <token>`, where token is [JSON Web Token](https://jwt.io/)
+authorizing the request.
+
+Client must specify the following claims:
+
+- `iat` (the time at which the JWT was issued
+  [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (for example
+  `1530644093`)
+- `exp` (the expiration time on or after which the JWT must not be accepted for processing,
+  [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4))
+- `host` domain of the server to which the request is being made for.
+- `path` the path of the request (`<WEB_AUTH_ENDPOINT>`)
+- All request's parameters (if specified in the request)
+
+Client must then correctly sign the payload with appropriate Stellar private key. To choose the private key client
+application should follow this steps:
+
+- If `client_domain` is specified, the token must be signed with the `SIGNING_KEY`. (See
```

Comment:
Lets use the term we defined in the beginning, `*Client Domain Account*` to make sure readers don't confuse the server and client's `SIGNING_KEY`.